common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataFlowTaskLaunchRequestAutoConfiguration.java

```diff
@@ -0,0 +1,125 @@
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeTypeUtils;
+
+/**
+ * @author David Turanski
+ **/
+@Configuration
+@EnableConfigurationProperties(DataflowTaskLaunchRequestProperties.class)
+public class DataFlowTaskLaunchRequestAutoConfiguration {
+
+	private static final Log log = LogFactory.getLog(TaskLaunchRequestTransformer.class);
+
+	private final DataflowTaskLaunchRequestProperties taskLaunchRequestProperties;
+
+	public DataFlowTaskLaunchRequestAutoConfiguration(DataflowTaskLaunchRequestProperties taskLaunchRequestProperties) {
+		this.taskLaunchRequestProperties = taskLaunchRequestProperties;
+	}
+
+	@Bean
+	public TaskLaunchRequestTransformer taskLaunchRequestTransformer() {
+		return message -> dataflowTaskLaunchRequest(message);
+	}
+
+	private Message dataflowTaskLaunchRequest(Message message) {
+
+		Assert.hasText(taskLaunchRequestProperties.getTaskName(), "'taskName' is required");
+		log.info(String.format("creating a task launch request for task %s", taskLaunchRequestProperties
+			.getTaskName()));
+		TaskLaunchRequestContext taskLaunchRequestContext = taskLaunchRequestContext(message);
+
+		DataFlowTaskLaunchRequest taskLaunchRequest = new DataFlowTaskLaunchRequest();
+		taskLaunchRequest.setCommandlineArguments(
+			taskLaunchRequestContext.mergeCommandLineArgs(taskLaunchRequestProperties));
+		taskLaunchRequest.setDeploymentProperties(
+			DeploymentPropertiesParser.parseDeploymentProperties(taskLaunchRequestProperties));
+		taskLaunchRequest.setTaskName(taskLaunchRequestProperties.getTaskName());
+		MessageBuilder<?> builder = MessageBuilder.withPayload(taskLaunchRequest).copyHeaders(message.getHeaders());
+
+		return adjustHeaders(builder, message.getHeaders()).build();
+	}
+
+	private MessageBuilder<?> adjustHeaders(MessageBuilder<?> builder, MessageHeaders messageHeaders) {
+		builder.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.APPLICATION_JSON);
+		if (messageHeaders.containsKey(TaskLaunchRequestContext.HEADER_NAME)) {
+			builder.removeHeader(TaskLaunchRequestContext.HEADER_NAME);
+		}
+		return builder;
+	}
+
+	private TaskLaunchRequestContext taskLaunchRequestContext(Message<?> message) {
+		TaskLaunchRequestContext taskLaunchRequestContext = (TaskLaunchRequestContext) message.getHeaders()
+			.get(TaskLaunchRequestContext.HEADER_NAME);
+
+		return taskLaunchRequestContext != null ? taskLaunchRequestContext : new TaskLaunchRequestContext();
+	}
+
+	public static class DataFlowTaskLaunchRequest {
+		@JsonProperty("args")
+		private List<String> commandlineArguments = new ArrayList<>();
+		@JsonProperty("deploymentProps")
```

Comment:
Would be better to reformat the way to ensure that each class member is surrounded with blank lines.

Reply:
Weird I just fixed this. I'm not sure what happened.